EncryptedStorage.py

```diff
@@ -0,0 +1,120 @@
+   try:
+      storage = EncryptedStorage(key=float(7), secret_value='LaZanzara')
+      assert False, f'Instantiation must fail if key is not an int. Instantiation successful with key of type float.'
+   except ValueError:
+      pass # A ValueError must be raised if key is not int.
+   
+   # Testing a ValueError is thrown if the key is of the wrong type: testing for str.
+   try:
+      storage = EncryptedStorage(key='7', secret_value='LaZanzara')
+      assert False, f'Instantiation must fail if key is not an int. Instantiation successful with key of type str.'
+   except ValueError:
+      pass # A ValueError must be raised if key is not int.
+   
+   # Testing a ValueError is thrown if the key is outside of the range [1, 25].
+   try:
+      storage = EncryptedStorage(key=0, secret_value='LaZanzara')
+      assert False, f'Instantiation must fail if key < 1. Instantiation successful with key = 0.'
+   except ValueError:
+      pass # A ValueError must be raised if key is < 1.
+      
+
+
+def test_access_secret_value():
+   storage = EncryptedStorage(key=7, secret_value='LaZanzara')
+   
+   # Checking that we are able to recover the secret value with the correct key.
+   val_correct_key = storage.access_secret_value(key=7)
+   assert val_correct_key == 'LaZanzara', f'''Problem recovering secret value, expected: 'LaZanzara', got '{val_correct_key}'.'''
+   
+   # Checking that the wrong key is giving out a different value of the correct length.
+   val_wrong_key_1 = storage.access_secret_value(key=3)
+   assert val_wrong_key_1 != 'LaZanzara', f'''The wrong key allows to return the secret value. Encryption key: 7, decryption key: 3'''
+   assert len(val_wrong_key_1) == len('LaZanzara'), f'''Using the wrong key returns a string with unexpected length. Got: {len(val_wrong_key_1)}.'''
+   
+   val_wrong_key_2 = storage.access_secret_value(key=4)
+   assert val_wrong_key_2 != 'LaZanzara', f'''The wrong key allows to return the secret value. Encryption key: 7, decryption key: 4.'''
+   assert len(val_wrong_key_2) == len('LaZanzara'), f'''Using the wrong key returns a string with unexpected length. Got: {len(val_wrong_key_2)}.'''
+   
+   # Checking that using different wrong keys is returning different outputs.
+   assert val_wrong_key_1 != val_wrong_key_2, f'''Using different keys produces the same output. Keys: 3, 4. Output: '{val_wrong_key_1}'.'''
+   
+
+def test_update_key():
+   storage = EncryptedStorage(key=2, secret_value='abc')
+   storage.update_key(old_key=2, new_key=3)
+   recovered_value_correct_key = storage.access_secret_value(key=3)
+   assert recovered_value_correct_key == 'abc', f'''Problem recovering secret value, expeted 'dfg', got '{recovered_value_correct_key}'.'''
+   
+   recovered_value_with_old_key = storage.acces_secret_value(key=3)
+   assert recovered_value_with_old_key != 'abc', 'Old key must not return correct secret value, expected different value'
+   
+   storage.update_key(old_key=1,new_key=4)
```

Comment:
Please add a comment specifying what is being tested here. Without a comment this code is not readable.